### PR TITLE
ESM build

### LINKED ISSUE #63: ESM build?
**Is your feature request related to a problem? Please describe.**
When including `attr-accept` in an ESM library, Vite fails to properly convert from CJS to ESM. See https://github.com/soorria/solid-dropzone/issues/3 for a real-world example.

**Describe the solution you'd like**
Offer an ESM build parallel to the current CJS options. I believe a larger Webpack configuration can do this relatively easily.

**Describe alternatives you've considered**
Forking just to produce our own ESM build.

### PR DESCRIPTION
**Pull request by edemaine**
Replace old `dist/es/index.js` with `dist/index.mjs` that uses ESM syntax, and properly link to it via `module` and `exports`.

Fixes #63

I also updated Webpack because otherwise I couldn't get it to run on modern Node.

**What kind of change does this PR introduce?**
- [ ] Bug Fix
- [ ] Feature
- [ ] Refactoring
- [ ] Style
- [x] Build
- [ ] Chore
- [ ] Documentation
- [ ] CI

**Did you add tests for your changes?**
- [ ] Yes, my code is well tested
- [x] Not relevant

**If relevant, did you update the documentation?**
- [x] Yes, I've updated the documentation
- [ ] Not relevant

**Summary**
See #63. This lets you `import accept from 'attr-accept'` in ESM code.

**Does this PR introduce a breaking change?**
If anyone directly imported `dist/es/index.js`, it no longer exists. (It seems to be a holdover that made a "fake" ES module.)

**Other information**
You can test the example in the README by making a file `test.mjs` and then running `node test.mjs`. (The `.mjs` extension forces ESM mode.)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,13 +3,17 @@
   "description": "JavaScript implementation of the \"accept\" attribute for HTML5 <input type=\"file\">",
   "version": "0.0.0-development",
   "main": "dist/index.js",
-  "module": "dist/es/index.js",
+  "module": "src/index.js",
+  "exports": {
+    "import": "./dist/index.mjs",
```
Comment (ArnaudBarre) [inline package.json:8]: You should provide types in the export map: https://publint.dev/rules#types_not_exported

**Comment by rolandjitsu** [comment]
Superseded by https://github.com/react-dropzone/attr-accept/pull/66?

File: package.json (hunk 1 above)
Comment (edemaine) [reply package.json:9]: Added! Sorry for the delay.

**Comment by edemaine** [comment]
I updated the deps in `package-lock.json`, as suggested in https://github.com/react-dropzone/attr-accept/pull/66#issuecomment-2403171182

This let me actually try `npm test`ing, and there were some additional issues that needed resolving (now fixed):
* Added `type: "module"` to `package.json` so that tests are treated as ESM. All files were already ESM anyway, except:
* Rename `webpack.config.js` to `webpack.config.cjs` so it can remain in CJS.
* Updated Mocha to enable ESM support.

If you want this PR and would rather it use a modern/faster build framework such as esbuild instead of Webpack, let me know.

**Comment by jonkoops** [comment]
This one can be closed now that #75 was merged.

**Pull request closed without merging**